invoices/sql_migration.go

```diff
@@ -358,3 +373,164 @@ func OverrideInvoiceTimeZone(invoice *Invoice) {
 		}
 	}
 }
+
+// MigrateInvoicesToSQL runs the migration of all invoices from the KV database
+// to the SQL database. The migration is done in a single transaction to ensure
+// that all invoices are migrated or none at all. This function can be run
+// multiple times without causing any issues as it will check if the migration
+// has already been performed.
+func MigrateInvoicesToSQL(ctx context.Context, db kvdb.Backend,
+	kvStore InvoiceDB, sqlStore *SQLStore, batchSize int) error {
+
+	offset := uint64(0)
+	var ops SQLInvoiceQueriesTxOptions
+	return sqlStore.db.ExecTx(ctx, &ops, func(tx SQLInvoiceQueries) error {
+		// First make sure that we haven't already migrated the
+		// invoices.
+		migration, err := tx.GetMigration(ctx, migrationID)
+		if err != nil {
+			return fmt.Errorf("unable to fetch migration '%v': %w",
+				migrationID, err)
+		}
+
+		// If there's a valid migration timestamp, then we've already
+		// migrated the invoices.
+		if migration.MigrationTs.Valid {
+			log.Infof("Migration '%v' has already been "+
+				"executed at %v", migrationID,
+				migration.MigrationTs.Time)
+
+			return nil
+		}
+
+		// Now create the hash index which we will use to look up
+		// invoice payment hashes by their add index during migration.
+		err = createInvoiceHashIndex(ctx, db, tx)
```

Comment:
we probably need to test it on a bigger db to make sure we do not run into memory issue having all this in memory

Reply:
Agreed. Do you think this should also be part of the PR and submitted to the repo? I think we should avoid bloating our repo too much.

Reply:
SGTM!